inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -572,22 +572,65 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::maneuvers_to_points(const std:
 
     auto lanelets = wm->getLaneletsBetween(starting_downtrack, lane_following_maneuver.end_dist, true);
 
+    if (lanelets.empty())
+    {
+      ROS_ERROR_STREAM("Detected no lanelets between starting_downtrack: " << starting_downtrack  << ", and lane_following_maneuver.end_dist: " << lane_following_maneuver.end_dist);
+      throw std::invalid_argument("Detected no lanelets between starting_downtrack and end_dist");
+    }
+
     ROS_DEBUG_STREAM("Maneuver");
 
     lanelet::BasicLineString2d downsampled_centerline;
     downsampled_centerline.reserve(200);
 
-    for (auto l : lanelets)
+    // getLaneletsBetween returns shortest path that is ordered
+    // exclude lanechanges and plan for only the straight part
```

Comment:
Can we add a comment to explain how/why a lane change may exist within inlanecruising plugin's maneuvers_to_points() function?

Reply:
Thanks! I added extra comments to clarify